### DISCUSSION TITLE
Writing Mockito based junit test cases

### DISCUSSION THREAD

**Discussion by ParthaNath28** [speedment/jpa-streamer#197] [Q&A]
Hello 

I has hoping to use mock test cases in my service layer just the way jpaRepository test cases are written. 

@Test
    @DisplayName("Get all list")
    public void testMyService() throws Exception {
        when(jpaStreamer.stream(MyEntity.class)
                .skip(20)
                .limit(20)
                .sorted((MyEntity.class)$.identifier)
                .toList()).thenReturn(myEntityList);

The error message I got is like below. Just wondering if there is a documentation available for using in Mock based test cases

org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 
ArrayList cannot be returned by stream()
stream() should return Stream
***

**Comment by minborg** [comment]
I am not an expert on mocking but I suspect that you only can have one layer of calls in the `when()` clause. So try and move the statements to the `thenReturn()` instead.
